### PR TITLE
Geos 3.9 Update

### PR DESCRIPTION
**Pull request by amystamile-usgs**
## Description
Updates GEOS from version 3.7 to 3.9

## Related Issue

#3627 

## How Has This Been Validated?

Ran tests through Jenkins and fixed all failing tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Infrastructure change (changes to things like CI or the build system that do not impact users)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [x] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by amystamile-usgs** [comment]
Still working on the isisminer stereopair2 test failure. Currently all csv have slightly different WKB results but all other data is correct. ~~When running the test with outputting WKTs instead of WKB the results have a 0.01 difference~~. See results below. 

WKT result from DEV (geos3.7)
`POLYGON ((239.9449041706754713 34.9703140706499980, 239.9552437504887621 34.9900335823538384, 239.9665152033978188 35.0115162333832899, 239.9777920172261929 35.0329949773491407, 239.9890741544317336 35.0544697284289413, 240.0003615774140826 35.0759404009236633, 240.0116542485167486 35.0974069092616503, 240.0229521300288980 35.1188691680027176, 240.0342551841877423 35.1403270918421811, 240.0455633731796752 35.1617805956148928, 240.0568766591428869 35.1832295942991351, 240.0681950041690698 35.2046740030208554, 240.0795183703051805 35.2261137370572897, 240.0908467195556852 35.2475487118412047, 240.1021800138841513 35.2689788429646498, 240.1135182152155494 35.2904040461829425, 240.1248612854378166 35.3118242374185130, 240.1362091864041872 35.3332393327648333, 240.1475618799346705 35.3546492484902117, 240.1589193278183529 35.3760539010416508, 240.1702814918152740 35.3974532070485850, 240.1816483336580177 35.4188470833267672, 240.1930198150542992 35.4402354468819780, 240.2043958976883289 35.4616182149137344, 240.2157765432227166 35.4829953048190276, 240.2271617133009727 35.5043666341960176, 240.2385513695488157 35.5257321208477350, 240.2499454735764743 35.5470916827856058, 240.2613439869803642 35.5684452382332026, 240.2727468713452481 35.5897927056297476, 240.2841540882458276 35.6111340036336870, 240.2955655992490165 35.6324690511262574, 240.3069813659156466 35.6537977672149680, 240.3184013498022580 35.6751200712370888, 240.3298255124633442 35.6964358827631330, 240.3412538154527454 35.7177451216002240, 240.3526862203261771 35.7390477077955282, 240.3641226886425386 35.7603435616396581, 240.3755631819659868 35.7816326036699550, 240.3870076618679832 35.8029147546738287, 240.3984560899289420 35.8241899356920399, 240.4099084277401346 35.8454580680219053, 240.4213646369057358 35.8667190732206222, 240.4328246790443302 35.8879728731083389, 240.4442885157910155 35.9092193897713798, 240.4557561087992212 35.9304585455653864, 240.4672274197424429 35.9516902631183726, 240.4787024103160888 35.9729144653337443, 240.4901810422393282 35.9941310753934971, 240.5016632772569665 36.0153400167609945, 240.5045794414372438 36.0181104469504163, 240.5312297422944994 36.0089386965450586, 240.5577378874835972 35.9997943329262498, 240.5842363661649017 35.9906318074573832, 240.6107250554471193 35.9814511569977569, 240.6372038326080087 35.9722524188580906, 240.6636725751005201 35.9630356307989985, 240.6901311605593605 35.9538008310295041, 240.7165794668069339 35.9445480582055268, 240.7430173718598496 35.9352773514282831, 240.7694447539350335 35.9259887502425741, 240.7958614914561792 35.9166822946352227, 240.8222674630596600 35.9073580250333464, 240.8486625476010659 35.8980159823025105, 240.8499312064927835 35.8975659244356109, 240.8228782911263011 35.8495056787270059, 240.7923640858847136 35.7951947501217376, 240.7618720824869456 35.7408216667379932, 240.7314020371914580 35.6863860023297050, 240.7009537062839115 35.6318873278659183, 240.6705268460648313 35.5773252114995699, 240.6401212128368456 35.5226992185361254, 240.6097365628928060 35.4680089114015828, 240.5793726525027978 35.4132538496102853, 240.5490292379016637 35.3584335897321225, 240.5187060752764694 35.3035476853594403, 240.4884029207537708 35.2485956870734825, 240.4581195303865684 35.1935771424103905, 240.4278556601416312 35.1384915958267072, 240.3976110658865935 35.0833385886645175, 240.3673855033767381 35.0281176591161341, 240.3371787282421508 34.9728283421881798, 240.3069904959742473 34.9174701696653713, 240.2768205619127855 34.8620426700736274, 240.2640150356526192 34.8435267606130310, 240.1987404796554983 34.8696009232497914, 240.1344018370785705 34.8952310415614519, 240.0700343285191991 34.9208030297532801, 240.0056376110311760 34.9463170302370543, 239.9449041706754713 34.9703140706499980))`

WKB result DEV

`0103000000010000005A000000F7DCABA73CFE6D402B256040337C4140D881575B91FE6D406313A16BB97E4140C6B54AB1EDFE6D408EE02A5D79814140D8087C124AFF6D40CB52EE2D398441404D53D77EA6FF6D40B61C33DDF8864140646548F602006E40C534416AB8894140A507BB785F006E4074D760D4778C414020FB1A06BC006E408689DA1A378F4140C1F9539E18016E403C1AF73CF691414078B6514175016E408DA5FF39B594414095DDFFEED1016E404A963D117497414004154AA72E026E4072A8FAC1329A41408BFC1B6A8B026E4030EB804BF19C4140192E6137E8026E4030C31AADAF9F4140FE3D050F45036E40AFEC12E66DA241403CBBF3F0A1036E40AC7DB4F52BA54140BE2F18DDFE036E4005E84ADBE9A74140AA205ED35B046E40A0FB2196A7AA4140950EB1D3B8046E4081E8852565AD4140D475FCDD15056E40E940C38822B04140BECE2BF272056E4064FB26BFDFB24140E48D2A10D0056E40EC74FEC79CB541406C24E4372D066E40F37297A259B84140410044698A066E406D25404E16BB4140568C35A4E7066E40DE2847CAD2BD4140FE30A4E844076E406288FB158FC041401A547B36A2076E40B6BFAC304BC341406A59A68DFF076E4028BDAA1907C64140C8A210EE5C086E40A7E345D0C2C841407490A557BA086E40B00CCF537ECB41404C8150CA17096E40458A97A339CE41401CD3FC4575096E40E328F1BEF4D04140DAE295CAD2096E406C312EA5AFD34140E50C0758300A6E40126BA1556AD6414054AD3BEE8D0A6E40411D9ECF24D9414027201F8DEB0A6E4079117812DFDB4140A0C19C34490B6E403195831D99DE414071EE9FE4A60B6E40B67B15F052E141400504149D040C6E40F91F83890CE44140C860E45D620C6E40656622E9C5E641406164FC26C00C6E40ADBE490E7FE94140F56F47F81D0D6E408F2550F837EC41406FE6B0D17B0D6E40A8268DA6F0EE4140B52C24B3D90D6E4024DE5818A9F14140F2A98C9C370E6E4084FA0B4D61F44140D6C7D58D950E6E4058BEFF4319F74140D3F2EA86F30E6E40ED018EFCD0F941405E9AB787510F6E40F634117688FC414030312790AF0F6E404E60E4AF3FFF4140882D25A00D106E407F2763A9F60142409FE6C88325106E401AAF707151024240283A84D5FF106E403AAA38E7240142406E4F20FDD8116E407B52BD42F9FF4140FDB97610B2126E4013B4E705CDFE4140C87F450F8B136E409FF60631A0FD4140FEBD4AF963146E40E8396BC472FC4140E2A944CE3C156E400C9565C044FB4140B291F18D15166E40AD15482516FA414078DD0F38EE166E401CBF65F3E6F84140EF0F5ECCC6176E407889122BB7F741405AC79A4A9F186E40BD60A3CC86F6414067BE84B277196E40E8236ED855F54140FFCCDA03501A6E4004A4C94E24F441402CE95B3E281B6E4026A30D30F2F2414090F7EDA2321B6E403CBAB170E3F241403A9FDA04551A6E400BF3219ABCEC4140CB6CED0B5B196E40A4DC0AF1C8E541408AC08F4161186E4069678F3ED3DE4140E2BE3EA567176E4019381C7FDBD74140FB8F77366E166E4071F817AFE1D04140085EB7F474156E400246E3CAE5C9414086537BDF7B146E40F3A0D8CEE7C241409A9940F682136E406A5A4CB7E7BB4140485684388A126E40DA828C80E5B44140BCAAC3A591116E40FBD7E026E1AD414091B17B3D99106E4099B28AA6DAA64140117D29FFA00F6E4021F4C4FBD19F41406A154AEAA80E6E40F0F3C322C7984140F0765AFEB00D6E405C6CB517BA9141405790D73AB90C6E408A67C0D6AA8A4140E2403E9FC10B6E40042C055C998341409F560B2BCA0A6E40FA289DA3857C41408D8CBBDDD2096E4059E29AA96F754140D388CBB6DB086E4083DC096A576E4140F8F8A8CF72086E40201155AFF86B41406090FE145C066E40F2F642154F6F4140CBD814054D046E406BEF46EE96724140CBD2A1B83D026E408D6CA9DFDC7541403560ED2E2E006E400C429CEA20794140F7DCABA73CFE6D402B256040337C4140`

WKT result Geos Branch (geos3.9)
`POLYGON ((239.9552437504887621 34.9900335823538384, 239.9665152033978188 35.0115162333832899, 239.9777920172261929 35.0329949773491407, 239.9890741544317336 35.0544697284289413, 240.0003615774140826 35.0759404009236633, 240.0116542485167486 35.0974069092616503, 240.0229521300288980 35.1188691680027176, 240.0342551841877423 35.1403270918421811, 240.0455633731796752 35.1617805956148928, 240.0568766591428869 35.1832295942991351, 240.0681950041690698 35.2046740030208554, 240.0795183703051805 35.2261137370572897, 240.0908467195556852 35.2475487118412047, 240.1021800138841513 35.2689788429646498, 240.1135182152155494 35.2904040461829425, 240.1248612854378166 35.3118242374185130, 240.1362091864041872 35.3332393327648333, 240.1475618799346705 35.3546492484902117, 240.1589193278183529 35.3760539010416508, 240.1702814918152740 35.3974532070485850, 240.1816483336580177 35.4188470833267672, 240.1930198150542992 35.4402354468819780, 240.2043958976883289 35.4616182149137344, 240.2157765432227166 35.4829953048190276, 240.2271617133009727 35.5043666341960176, 240.2385513695488157 35.5257321208477350, 240.2499454735764743 35.5470916827856058, 240.2613439869803642 35.5684452382332026, 240.2727468713452481 35.5897927056297476, 240.2841540882458276 35.6111340036336870, 240.2955655992490165 35.6324690511262574, 240.3069813659156466 35.6537977672149680, 240.3184013498022580 35.6751200712370888, 240.3298255124633442 35.6964358827631330, 240.3412538154527454 35.7177451216002240, 240.3526862203261771 35.7390477077955282, 240.3641226886425386 35.7603435616396581, 240.3755631819659868 35.7816326036699550, 240.3870076618679832 35.8029147546738287, 240.3984560899289420 35.8241899356920399, 240.4099084277401346 35.8454580680219053, 240.4213646369057358 35.8667190732206222, 240.4328246790443302 35.8879728731083389, 240.4442885157910155 35.9092193897713798, 240.4557561087992212 35.9304585455653864, 240.4672274197424429 35.9516902631183726, 240.4787024103160888 35.9729144653337443, 240.4901810422393282 35.9941310753934971, 240.5016632772569665 36.0153400167609945, 240.5045794414372438 36.0181104469504163, 240.5312297422944994 36.0089386965450586, 240.5577378874835972 35.9997943329262498, 240.5842363661649017 35.9906318074573832, 240.6107250554471193 35.9814511569977569, 240.6372038326080087 35.9722524188580906, 240.6636725751005201 35.9630356307989985, 240.6901311605593605 35.9538008310295041, 240.7165794668069339 35.9445480582055268, 240.7430173718598496 35.9352773514282831, 240.7694447539350335 35.9259887502425741, 240.7958614914561792 35.9166822946352227, 240.8222674630596600 35.9073580250333464, 240.8486625476010659 35.8980159823025105, 240.8499312064927835 35.8975659244356109, 240.8228782911263011 35.8495056787270059, 240.7923640858847136 35.7951947501217376, 240.7618720824869456 35.7408216667379932, 240.7314020371914580 35.6863860023297050, 240.7009537062839115 35.6318873278659183, 240.6705268460648313 35.5773252114995699, 240.6401212128368456 35.5226992185361254, 240.6097365628928060 35.4680089114015828, 240.5793726525027978 35.4132538496102853, 240.5490292379016637 35.3584335897321225, 240.5187060752764694 35.3035476853594403, 240.4884029207537708 35.2485956870734825, 240.4581195303865684 35.1935771424103905, 240.4278556601416312 35.1384915958267072, 240.3976110658865935 35.0833385886645175, 240.3673855033767381 35.0281176591161341, 240.3371787282421508 34.9728283421881798, 240.3069904959742473 34.9174701696653713, 240.2768205619127855 34.8620426700736274, 240.2640150356526192 34.8435267606130310, 240.1987404796554983 34.8696009232497914, 240.1344018370785705 34.8952310415614519, 240.0700343285191991 34.9208030297532801, 240.0056376110311760 34.9463170302370543, 239.9449041706754713 34.9703140706499980, 239.9552437504887621 34.9900335823538384))`

WKB result Geos Branch
`0103000000010000005A000000D881575B91FE6D406313A16BB97E4140C6B54AB1EDFE6D408EE02A5D79814140D8087C124AFF6D40CB52EE2D398441404D53D77EA6FF6D40B61C33DDF8864140646548F602006E40C534416AB8894140A507BB785F006E4074D760D4778C414020FB1A06BC006E408689DA1A378F4140C1F9539E18016E403C1AF73CF691414078B6514175016E408DA5FF39B594414095DDFFEED1016E404A963D117497414004154AA72E026E4072A8FAC1329A41408BFC1B6A8B026E4030EB804BF19C4140192E6137E8026E4030C31AADAF9F4140FE3D050F45036E40AFEC12E66DA241403CBBF3F0A1036E40AC7DB4F52BA54140BE2F18DDFE036E4005E84ADBE9A74140AA205ED35B046E40A0FB2196A7AA4140950EB1D3B8046E4081E8852565AD4140D475FCDD15056E40E940C38822B04140BECE2BF272056E4064FB26BFDFB24140E48D2A10D0056E40EC74FEC79CB541406C24E4372D066E40F37297A259B84140410044698A066E406D25404E16BB4140568C35A4E7066E40DE2847CAD2BD4140FE30A4E844076E406288FB158FC041401A547B36A2076E40B6BFAC304BC341406A59A68DFF076E4028BDAA1907C64140C8A210EE5C086E40A7E345D0C2C841407490A557BA086E40B00CCF537ECB41404C8150CA17096E40458A97A339CE41401CD3FC4575096E40E328F1BEF4D04140DAE295CAD2096E406C312EA5AFD34140E50C0758300A6E40126BA1556AD6414054AD3BEE8D0A6E40411D9ECF24D9414027201F8DEB0A6E4079117812DFDB4140A0C19C34490B6E403195831D99DE414071EE9FE4A60B6E40B67B15F052E141400504149D040C6E40F91F83890CE44140C860E45D620C6E40656622E9C5E641406164FC26C00C6E40ADBE490E7FE94140F56F47F81D0D6E408F2550F837EC41406FE6B0D17B0D6E40A8268DA6F0EE4140B52C24B3D90D6E4024DE5818A9F14140F2A98C9C370E6E4084FA0B4D61F44140D6C7D58D950E6E4058BEFF4319F74140D3F2EA86F30E6E40ED018EFCD0F941405E9AB787510F6E40F634117688FC414030312790AF0F6E404E60E4AF3FFF4140882D25A00D106E407F2763A9F60142409FE6C88325106E401AAF707151024240283A84D5FF106E403AAA38E7240142406E4F20FDD8116E407B52BD42F9FF4140FDB97610B2126E4013B4E705CDFE4140C87F450F8B136E409FF60631A0FD4140FEBD4AF963146E40E8396BC472FC4140E2A944CE3C156E400C9565C044FB4140B291F18D15166E40AD15482516FA414078DD0F38EE166E401CBF65F3E6F84140EF0F5ECCC6176E407889122BB7F741405AC79A4A9F186E40BD60A3CC86F6414067BE84B277196E40E8236ED855F54140FFCCDA03501A6E4004A4C94E24F441402CE95B3E281B6E4026A30D30F2F2414090F7EDA2321B6E403CBAB170E3F241403A9FDA04551A6E400BF3219ABCEC4140CB6CED0B5B196E40A4DC0AF1C8E541408AC08F4161186E4069678F3ED3DE4140E2BE3EA567176E4019381C7FDBD74140FB8F77366E166E4071F817AFE1D04140085EB7F474156E400246E3CAE5C9414086537BDF7B146E40F3A0D8CEE7C241409A9940F682136E406A5A4CB7E7BB4140485684388A126E40DA828C80E5B44140BCAAC3A591116E40FBD7E026E1AD414091B17B3D99106E4099B28AA6DAA64140117D29FFA00F6E4021F4C4FBD19F41406A154AEAA80E6E40F0F3C322C7984140F0765AFEB00D6E405C6CB517BA9141405790D73AB90C6E408A67C0D6AA8A4140E2403E9FC10B6E40042C055C998341409F560B2BCA0A6E40FA289DA3857C41408D8CBBDDD2096E4059E29AA96F754140D388CBB6DB086E4083DC096A576E4140F8F8A8CF72086E40201155AFF86B41406090FE145C066E40F2F642154F6F4140CBD814054D046E406BEF46EE96724140CBD2A1B83D026E408D6CA9DFDC7541403560ED2E2E006E400C429CEA20794140F7DCABA73CFE6D402B256040337C4140D881575B91FE6D406313A16BB97E4140`

**Comment by amystamile-usgs** [comment]
Looking more at the comparison of WKB/WKT strings, they both contain the same results but they start and stop at different values. But the overall polygon values are the same.

Here is the comparison showing that the first WKB starts with `0F7DCABA73CFE6D402B256040337C414` or `(239.9449041706754713 34.9703140706499980)` and the second WKB ends with  `239.9449041706754713 34.9703140706499980` or `(239.9552437504887621 34.9900335823538384)`
image

**Comment by thareUSGS** [comment]
Just for fun, I ran a Symmetrical difference between the two polygons in QGIS, and a NULL polygon was returned. That just means it is the same polygon. I think your testing is complete and all looks good.

https://docs.qgis.org/3.16/en/docs/user_manual/processing_algs/qgis/vectoroverlay.html#qgissymmetricaldifference

**Comment by acpaquette** [comment]
Thank you custom ISIS test evaluation!

File: isis/src/base/objs/Chip/unitTest.cpp [hunk 1]
```diff
@@ -134,15 +134,15 @@ int main() {
...
 
-  geos::geom::CoordinateSequence *pts = new geos::geom::CoordinateArraySequence();
+  geos::geom::CoordinateArraySequence *pts = new geos::geom::CoordinateArraySequence();
   pts->add(geos::geom::Coordinate(23.0, 22.0));
   pts->add(geos::geom::Coordinate(28.0, 22.0));
   pts->add(geos::geom::Coordinate(28.0, 27.0));
   pts->add(geos::geom::Coordinate(25.0, 28.0));
   pts->add(geos::geom::Coordinate(23.0, 22.0));
-  vector<geos::geom::Geometry *> polys;
+  vector<geos::geom::Geometry *> *polys = new vector<geos::geom::Geometry *>;
```
Comment (AustinSanders) [inline isis/src/base/objs/Chip/unitTest.cpp:143]: I think this results in a memory leak.  We could use a smart pointer via std::make_unique or std::make_shared or just track it down and delete the memory

File: isis/src/base/objs/GridPolygonSeeder/unitTest.cpp [hunk 2]
```diff
@@ -68,8 +68,8 @@ int main() {
       cout << "Test 2, test a square polygon" << endl;
       try {
         // Call the seed member with a polygon
-        geos::geom::CoordinateSequence *pts;
-        vector<geos::geom::Geometry *> polys;
+        geos::geom::CoordinateArraySequence *pts;
+        auto *polys = new vector<geos::geom::Geometry *>;
```
Comment (AustinSanders) [inline isis/src/base/objs/GridPolygonSeeder/unitTest.cpp:72]: Same issue with memory leak

File: isis/src/base/objs/PolygonTools/PolygonTools.cpp [hunk 3]
```diff
@@ -13,14 +13,16 @@ find files of those names at the top level of this repository. **/
 #include <cmath>
 
 #include <QDebug>
-#include "geos/geom/BinaryOp.h"
+
+//#include "geos/geom/BinaryOp.h"
```
Comment (AustinSanders) [inline isis/src/base/objs/PolygonTools/PolygonTools.cpp]: Can this be removed or documented why it's commented out?

**Review by AustinSanders — CHANGES_REQUESTED** [review]
A couple minor changes.  I could be wrong on the memory leaks

File: isis/src/base/objs/Chip/unitTest.cpp (hunk 1 above)
Comment (amystamile-usgs) [reply isis/src/base/objs/Chip/unitTest.cpp:143]: I added `delete polys;` but this caused both tests to fail.

Comment (AustinSanders) [reply isis/src/base/objs/Chip/unitTest.cpp:143]: That's interesting... is there something that I'm missing with the heap allocation here?  I'd expect it to need deallocation somewhere unless `polys` is part of an object with its own destructors.

Maybe smart pointers would manage this for us?  A memory leak in the tests isn't great, but it's also not the end of the world. @Kelvinrr @acpaquette any thoughts?

Comment (acpaquette) [reply isis/src/base/objs/Chip/unitTest.cpp:143]: ISIS handles pointers really poorly and likely needs to have verification that pointers get cleaned up. In this case the chip classes destruction deletes the pointer see here:
https://github.com/DOI-USGS/ISIS3/blob/1e21b588f16096b1a7ad4e7ddd597ba1500a876e/isis/src/base/objs/Chip/Chip.cpp#L89

Comment (AustinSanders) [reply isis/src/base/objs/Chip/unitTest.cpp:143]: Good catch!  This should be fine then

**Review by AustinSanders — APPROVED** [review]

**Pull request merged**